flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSource.java

```diff
@@ -111,24 +112,17 @@
 		return kafkaSource;
 	}
 
-	@Override
-	public int getNumberOfFields() {
-		return fieldNames.length;
-	}
-
-	@Override
 	public String[] getFieldsNames() {
-		return fieldNames;
+		return TableSource$class.getFieldsNames(this);
```

Comment:
`CsvTableSource` is inheriting both `StreamTableSource` and `BatchTableSource` so they should be traits.

I don't think that adding a method that only calling implementation from a trait is a big issue. In any case we do not duplicate the code and do not re-implement methods.

Do you have any concerns about this?